debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java

```diff
@@ -201,37 +203,20 @@ protected int createRecords(Struct source, Map<String, ?> offset, Operation oper
             return 1;
         }
 
-        protected String objectIdLiteralFrom(Document obj) {
-            if (obj == null) {
+        protected String idObjToJson(Object idObj) {
+            if(idObj == null) {
                 return null;
             }
-            Object id = obj.get("_id");
-            return objectIdLiteral(id);
-        }
-
-        protected String objectIdLiteral(Object id) {
-            if (id == null) {
-                return null;
-            }
-            if (id instanceof ObjectId) {
-                return ((ObjectId) id).toHexString();
-            }
-            if (id instanceof String) {
-                return (String) id;
-            }
-            if (id instanceof Document) {
-                Document doc = (Document)id;
-                if (doc.containsKey("_id") && doc.size() == 1) {
-                    // This is an embedded ObjectId ...
-                    return objectIdLiteral(doc.get("_id"));
-                }
-                return valueTransformer.apply((Document) id);
+            if(!(idObj instanceof Document)) {
+                return JSONSerializers.getStrict().serialize(idObj);
```

Comment:
I'm a bit concerned about getting a new serializer each time. Would be nice to do it only once and store as a field in `RecordMakers`, but then `JSONSerializers` doesn't make any guarantees on thread-safety. Worse, `ClassMapBasedObjectSerializer` says it's not thread-safe. OTOH, looking at its code it seems safe to me, given we won't add any serializers after we have retrieved it. So we could try and store it as a field indeed. WDYT?

Reply:
yeah! definitely makes sense to avoid recreating this object if thread-safety can be ignored. I'll change it like you suggest.